src/utils/hooks/useGetFullReviewStructure.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from 'react'
+import { FullStructure } from '../types'
+import {
+  ApplicationResponse,
+  Review,
+  ReviewAssignment,
+  ReviewResponse,
+  useGetReviewNewQuery,
+} from '../generated/graphql'
+import addEvaluatedResponsesToStructure from '../helpers/structure/addEvaluatedResponsesToStructure'
+import { useUserState } from '../../contexts/UserState'
+import addOwnedReviewResponse from '../helpers/structure/addOwnedReviewResponse'
+import addElementsById from '../helpers/structure/addElementsById'
+
+interface useGetFullApplicationStructureProps {
+  structure: FullStructure
+  reviewAssignment: ReviewAssignment
+}
+
+const useGetFullReviewStructure = ({
+  structure,
+  reviewAssignment,
+}: useGetFullApplicationStructureProps) => {
+  const [fullStructure, setFullStructure] = useState<FullStructure>()
+  const [isError, setIsError] = useState(false)
+
+  const {
+    userState: { currentUser },
+  } = useUserState()
+
+  const { data, error } = useGetReviewNewQuery({
+    variables: {
+      reviewAssignmentId: reviewAssignment.id,
+      userId: currentUser?.userId || 0,
+    },
+  })
+
+  useEffect(() => {
+    if (error) {
+      setIsError(true)
+      return
+    }
+
+    if (!data) return
+
+    addEvaluatedResponsesToStructure({
+      structure,
+      applicationResponses: data?.reviewAssignment?.application?.applicationResponses
+        ?.nodes as ApplicationResponse[],
+      currentUser,
+      evaluationOptions: {
+        isEditable: false,
+        isVisible: true,
+        isRequired: false,
+        isValid: false,
+      },
+    }).then((evaluatedStructure: FullStructure) => {
+      let newStructure: FullStructure = addElementsById(evaluatedStructure)
+
+      // here we add reviewless review repsonses
```

Comment:
What would these be?

Reply:
like when consolidation is not started etc..

Reply:
Ahhh now I understood `reviewless` as without an associated review. But could it be that you get review_responses that doesn't have any review associated? Not right, they have some review it might not be yours...

Reply:
yep that's correct, my comment wasn't though 9dfd303d37e3b41e56cf11d32ec9de3f4897cee4